accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -901,6 +904,56 @@ These are extended as
 per
 [separate proposal](https://github.com/dart-lang/language/blob/master/resources/type-system/flow-analysis.md).
 
+### The main function
+
+The section 'Scripts' in the language specification is replaced by the
+following:
+
+Let _L_ be a library that exports a declaration _D_ named `main`.  It is a
+compile-time error unless _D_ is a function declaration.  It is a
```

Comment:
This excludes getters, right? co19/Language/Libraries_and_Scripts/Scripts/top_level_main_t05 seems to assume it doesn't.

Reply:
It is intended to exclude getters. The old version of this text in section `Scripts` said explicitly in commentary that we couldn't have a 'top-level getter or field named `main`', and we did not discuss nor intend to change that when this spec change was landed.

However, section 'Functions' implies that 'function' includes getters, setters, constructors, and "functions" (no, we don't have a separate term for the ones that have a `<formalParameterList>` and aren't constructors, I had a CL a long time ago to fix that, but it hasn't been landed). So the text should say `unless _D_ is a non-getter function declaration`. It can't be a top-level constructor and a setter can't have the name `main`, so that should be enough.

Reply:
https://github.com/dart-lang/language/pull/1146

Reply:
When calling `main` from external code, e.g. JavaScript, it is necessary to _convert_ arguments. We can't just say that `main` is called dynamically from 'outside'. This conversion happens in the context of the program but before entering `main`, e.g. needs to access the type `List<String>` for that program. This interstitial place a special place that we want to constrain heavily in a platform dependent way. For dart2js we want limit the conversion to just `Array`-to-`List<String>`, for which we need to know the static type of the entry point.

Reply:
@rakudrama does line 920 not address your concerns?

Reply:
Yes, it does. I was just adding colour for Johnni.